deltachat-ffi/src/providers.rs

```diff
@@ -1,93 +1,31 @@
 extern crate deltachat_provider_database;
 
-use std::ptr;
-
 use crate::string::{to_string_lossy, StrExt};
-use deltachat_provider_database::StatusState;
 
 #[no_mangle]
 pub type dc_provider_t = deltachat_provider_database::Provider;
 
 #[no_mangle]
-pub unsafe extern "C" fn dc_provider_new_from_domain(
+pub unsafe extern "C" fn dc_provider_json_from_domain(
     domain: *const libc::c_char,
-) -> *const dc_provider_t {
-    match deltachat_provider_database::get_provider_info(&to_string_lossy(domain)) {
-        Some(provider) => provider,
-        None => ptr::null(),
+) -> *mut libc::c_char {
+    let domain = to_string_lossy(domain);
+    match deltachat_provider_database::get_provider_info(&domain) {
+        Some(provider) => serde_json::to_string(provider).unwrap().strdup(),
```

Comment:
`.unwrap_or_default()` is more defensive than plain `.unwrap()`.  But even then you're silently swallowing an error, sadly `.unwrap_or_log_default()` or any other form of logging doesn't work as you don't have a context though.  Perhaps we do need to integrate the rust log crate in some way...

Reply:
what should be inside of default? empty string?

Reply:
`.unwrap_or_default()` uses the "Default" trait to use a standardised "default" value for the type.  In this case the type is String and the default for that is the empty string.  So you don't have to explicitly write the default value (if you wanted that you'd use `.unwrap_or_else("some_val")`).  Since returning the empty string is here the way to return an error code substituting `.unwrap()` with `.unwrap_or_default()` is all that should be needed.